packages/medusa/src/services/batch-job.ts

```diff
@@ -1,34 +1,99 @@
-import { EntityManager } from "typeorm"
-
+import { DeepPartial, EntityManager } from "typeorm"
```

Comment:
thought: would it make sense to rename the functions that will be called by the Strategies to something like `set*`.

E.g. when a strategy starts pre-processing it could call `batchService.setPreProcessing` instead of simply `batchService.preProcessing`.

I think this signifies what is happening a bit closer. Wdyt?

Result would be this:

```
complete -> setComplete
ready -> setReady
processing -> setProcessing
preProcessing -> setPreProcssing
```

Reply:
Entirely true 🚀